docs/Documentation/SystemDesign/4-StorageEngine/6-DataManipulation.md

```diff
@@ -0,0 +1,96 @@
+<!--
+
+    Licensed to the Apache Software Foundation (ASF) under one
+    or more contributor license agreements.  See the NOTICE file
+    distributed with this work for additional information
+    regarding copyright ownership.  The ASF licenses this file
+    to you under the Apache License, Version 2.0 (the
+    "License"); you may not use this file except in compliance
+    with the License.  You may obtain a copy of the License at
+    
+        http://www.apache.org/licenses/LICENSE-2.0
+    
+    Unless required by applicable law or agreed to in writing,
+    software distributed under the License is distributed on an
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+
+-->
+
+# Data addition, deletion and modification
+
+The following describes four common data manipulation operations, which are insert, update, delete, and TTL settings.
+
+## Data insertion
+
+### Write a single line of data (one device, one timestamp, multiple values)
+
+* Corresponding interface
+	* JDBC's execute and executeBatch interfaces
+	* Session's insert and insertInBatch
+
+* Main entrance: public void insert(InsertPlan insertPlan)   StorageEngine.java
+	* Find the corresponding StorageGroupProcessor
+	* Find the corresponding TsFileProcessor according to the time of writing the data and the last time stamp of the current device order
+	* Pre-write log
+	* Write to mestable corresponding to TsFileProcessor
+	    * If the file is out of order, update the endTimeMap in tsfileResource
+	    * If there is no information about the device in tsfile, then update the startTimeMap in tsfileResource
+	* Determine whether to trigger asynchronous persistent memtable operation based on memtable size
+	    * If it is a sequential file and the flashing action is performed, the endTimeMap in tsfileResource is updated
+	* Determine whether to trigger a file close operation based on the size of the current disk TsFile
+
+### Batch data (multiple timestamp multiple values for one device) write
+
+* Corresponding interface
+	* Session‘s insertBatch
+
+* Main entrance: public Integer[] insertBatch(BatchInsertPlan batchInsertPlan)  StorageEngine.java
```

Comment:
You may add code quote at signature of method. Just like `public Integer[] insertBatch(BatchInsertPlan batchInsertPlan)`